vllm/sequence.py

```diff
@@ -644,6 +649,9 @@ def get_encoder_seq(self) -> Optional[Sequence]:
         return self.encoder_seq
 
     def get_unfinished_seqs(self) -> List[Sequence]:
+        if self.is_single_seq:
+            return self.seqs if not self.seqs[0].is_finished() else []
+
         return [seq for seq in self.seqs if not seq.is_finished()]
 
     def get_finished_seqs(self) -> List[Sequence]:
```

Comment:
why no shortcut here?

Reply:
Good catch, addressed in https://github.com/vllm-project/vllm/pull/7364